app/Http/Controllers/AvatarController.php

```diff
@@ -0,0 +1,34 @@
+<?php
+
+namespace App\Http\Controllers;
+
+use Illuminate\Http\Request;
+use App\Http\Requests;
+use App\Http\Controllers\Controller;
+use App\Friend;
+use Illuminate\Support\Facades\File;
+use Symfony\Component\HttpFoundation\File\MimeType\ExtensionGuesser;
+use Symfony\Component\HttpFoundation\BinaryFileResponse;
+
+class AvatarController extends Controller
+{
+    public function show($username)
```

Comment:
Cache in what way? Realistically, we could probably just symlink the directory into `public` and serve the image directly. This was more around serving appropriate mime types, but I could do the lookup when I save the file; just need to tweak the way we determine if the user's avatar already exists (say if they change from a jpg to a png, the filename will still be the same, but the extension changes).